src/main/java/org/radarbase/output/config/RestructureConfig.kt

```diff
@@ -0,0 +1,281 @@
+package org.radarbase.output.config
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import io.minio.MinioClient
+import org.apache.hadoop.conf.Configuration
+import org.radarbase.output.Application.Companion.CACHE_SIZE_DEFAULT
+import org.radarbase.output.Plugin
+import org.radarbase.output.path.ObservationKeyPathFactory
+import org.radarbase.output.compression.Compression
+import org.radarbase.output.compression.CompressionFactory
+import org.radarbase.output.format.FormatFactory
+import org.radarbase.output.format.RecordConverterFactory
+import org.radarbase.output.path.RecordPathFactory
+import org.radarbase.output.storage.LocalStorageDriver
+import org.radarbase.output.storage.S3StorageDriver
+import org.radarbase.output.storage.StorageDriver
+import org.slf4j.LoggerFactory
+import java.lang.IllegalStateException
+import java.net.URI
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+data class RestructureConfig(
+        /** Whether and how to run as a service. */
+        val service: ServiceConfig = ServiceConfig(enable = false),
+        /** Work limits. */
+        val worker: WorkerConfig = WorkerConfig(),
+        /** Topic exceptional handling. */
+        val topics: Map<String, TopicConfig> = emptyMap(),
+        /** Source data resource configuration. */
+        val source: ResourceConfig = ResourceConfig("hdfs", hdfs = HdfsConfig()),
+        /** Target data resource configration. */
+        val target: ResourceConfig = ResourceConfig("local", local = LocalConfig()),
+        /** Redis configuration for synchronization and storing offsets. */
+        val redis: RedisConfig = RedisConfig(),
```

Comment:
Would it still be possible to write the locks how we used to? I think it was local file system? 
Or do we have to set-up redis when we upgrade?

Reply:
Redis is mandatory now. Locking happened in HDFS before, but since we plan on phasing that out. The code still exists to do the locking in HDFS, but it cannot be configured that way.